plugins/ag-charts-build-tools/src/executors/generate-chart-thumbnail/generator/environment.ts

```diff
@@ -0,0 +1,61 @@
+import { JSDOM } from 'jsdom';
+
+import { AgCharts } from 'ag-charts-community';
+import { mockCanvas } from 'ag-charts-test';
+
+import { DEFAULT_THUMBNAIL_HEIGHT, DEFAULT_THUMBNAIL_WIDTH, THUMBNAIL_POOL_SIZE } from './constants';
+
+function buildPoolEntry() {
+    const jsdom = new JSDOM('<html><head><style></style></head><body><div id="myChart"></div></body></html>');
+
+    const mockCtx = mockCanvas.setup({
+        width: DEFAULT_THUMBNAIL_WIDTH,
+        height: DEFAULT_THUMBNAIL_HEIGHT,
+        document: jsdom.window.document,
+        window: jsdom.window as any,
+        mockText: false,
+    });
+
+    const options = {
+        animation: { enabled: false },
+        document: jsdom.window.document,
+        window: jsdom.window,
+        width: DEFAULT_THUMBNAIL_WIDTH,
+        height: DEFAULT_THUMBNAIL_HEIGHT,
+    } as any;
+
+    const chartProxy = AgCharts.create(options);
+
+    return [jsdom, mockCtx, chartProxy] as const;
+}
+
+const pool: ReturnType<typeof buildPoolEntry>[] = [];
+
+function initPool() {
+    // eslint-disable-next-line no-console
+    console.log(`Creating thumbnail pool of size ${THUMBNAIL_POOL_SIZE}`);
+    for (let i = 0; i < THUMBNAIL_POOL_SIZE; i++) {
+        pool.push(buildPoolEntry());
+    }
+}
+initPool();
+
+export async function borrowFromPool() {
+    let count = 0;
+    while (pool.length === 0 && count < 5) {
```

Comment:
Should the `5` here be `THUMBNAIL_POOL_SIZE`?

```suggestion
    while (pool.length === 0 && count < THUMBNAIL_POOL_SIZE) {
```